MAINTAINERS.md

```diff
@@ -77,8 +85,54 @@ Make frequent project releases to the community.
 
 ### Promote Other Maintainers
 
-Assist, add, and remove [MAINTAINERS](MAINTAINERS.md). Exercise good judgement, and propose high quality contributors to become co-maintainers.
+Assist, add, and remove [MAINTAINERS](MAINTAINERS.md). Exercise good judgement, and propose high quality contributors to become co-maintainers. See [Becoming a Maintainer](#becoming-a-maintainer) for more information.
 
 ### Describe the Repo
 
 Make sure the repo has a well-written, accurate, and complete description. See [opensearch-project/.github#38](https://github.com/opensearch-project/.github/issues/38) for some helpful tips to describe your repo.
+
+## Becoming a Maintainer
+
+You can become a maintainer by actively [contributing](CONTRIBUTING.md) to any project, and being nominated by an existing maintainer.
+
+### Nomination
+
+Any current maintainer starts a private e-mail thread (until we have a better mechanism, e-mail addresses can usually be found via MAINTAINERS.md + DCO) with all other maintainers on that repository to discuss nomination using the template below. In order to be approved, at least three positive maintainer votes are necessary, and no vetoes. In rare cases when there are fewer than three maintainers, the positive votes from all maintainers are required. Any disagreements can be escalated to the repo admin.
+
+The nomination should clearly identify the person with their real name and a link to their GitHub profile, and the rationale for the nomination, with concrete example contributions.
+
+### Interest
+
+Upon receiving at least three positive maintainer votes, and no vetoes, from other maintainers within a maximum period of two weeks, the nominating maintainer asks a potential nominee whether they might be interested in becoming a maintainer on the repository, which happens via private e-mail message message, but may also be a comment to a pull request, following a substantial contribution.
+
+> This is great work! Based on your ongoing engagement with the project, the current maintainers now invite you to become a co-maintainer for this project. Please respond and let us know if you would like to become maintainer.
```

Comment:
```suggestion
> This is great work! Based on your valuable contribution and ongoing engagement with the project, the current maintainers invite you to become a co-maintainer for the OpenSearch project. Please respond and let us know if you accept the invitation to become maintainer.
```

Reply:
https://github.com/opensearch-project/.github/pull/59/commits/9580d11a4f1bbf39ade9cf0ba8d66bed2b40ffb7

I didn't keep the "the OpenSearch project" because this is generic for all repos and I don't want to misrepresent that the maintainer is only being added to a specific repo. I don't want to call the repo a repo though, we call them projects.